jardin/instrumentation/event.py

```diff
@@ -0,0 +1,10 @@
+from typing import *
+from dataclasses import dataclass, field
+
+@dataclass
+class Event:
+  name: str
+  start_time: float = 0.0
```

Comment:
needs a comment: seconds since unix epoch?

Reply:
Also, I would consider factoring out `start_time`, `end_time` and `duration` since they will either all be set or all be unset. Right now it's too easy to create an invalid configuration.

Maybe something like:
```python
@dataclass
class Event:
    name: str
    timing: Optional[EventTiming] = None
    tags: Dict[str, str] = field(default_factory=dict)

@dataclass
class EventTiming:
    start_time: float
    end_time: float
    duration: float
```

Reply:
And one more thing 😄  ...
Couldn't `end_time` be derived by adding `duration` to the `start_time`?